internal/api/auth_test.go

```diff
@@ -0,0 +1,197 @@
+package api
+
+import (
+	"bytes"
+	"errors"
+	"net/http"
+	"testing"
+
+	"gophermart/internal/mock"
+
+	"net/http/httptest"
+
+	gomock "github.com/golang/mock/gomock"
+)
+
+func setupHandler(t *testing.T) *Handler {
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockStore := mock.NewMockStore(ctrl)
+
+	return &Handler{store: mockStore}
+}
+
+type want struct {
+	contentType string
+	statusCode  int
+}
+
+type authTescases []struct {
+	name       string
+	method     string
+	url        string
+	reqBody    string
+	mockuserID int
+	mockHash   []byte
+	mockErr    error
+	want       want
+}
+
+func TestHandler_Register(t *testing.T) {
+	tests := authTescases{
+		{
+			name:   "register_status_code_200",
+			url:    "/api/user/register",
+			method: http.MethodPost,
+			want: want{
+				contentType: "application/json",
+				statusCode:  http.StatusOK,
+			},
+			reqBody: `{"login": "user", "password": "123456"}`,
+		},
+		{
+			name:   "register_status_code_400",
```

Comment:
Мне кажется вот такое иформативное имя намного лучшее поможет найти сломанный тест не говоря уже о том, что оно описывает то, что проверяется в тесте.

```suggestion
			name:   "register_with_invalid_json_body_should_repond_status_code_400",
```